### PR TITLE
fix: now the workspace ID cookie is set only when a valid workspace page is loaded

### LINKED ISSUE #2822: Bug: Non-existent pages for non-existent workspaces generate an invalid workspace ID cookie value
If you go to a valid page for a workspace but with an invalid workspace ID, the page errors and deletes the workspace ID cookie.

However if you go to an invalid page for a workspace with an invalid workspace ID, the page 404s and the workspace ID cookie gets set with an invalid workspace ID.

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Now the workspace ID cookie is only set when a workspace page successfully loads.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #2822

## Steps to QA

Ensure you're logged in.

1. Go to a non existent workspace page, https://deploy-preview-2835--oss-insights.netlify.app/workspaces/21c5e583-f1fd-40q3-9ec9-7801f67ff727/yolo
2. The page loads gives a 404
3. Go to a https://deploy-preview-2835--oss-insights.netlify.app/workspaces
5. The last accessed workspace loads fine.

You can compare beta with the same steps and you will get a crash when you go to /workspaces

1. Go to a non existent workspace page, e.g. https://beta.app.opensauced.pizza/workspaces/21c5e583-f1fd-40q3-9ec9-7801f67ff727/yolo
2. The page loads gives a 404
3. Go to a https://beta.app.opensauced.pizza/workspaces
4. The site gives an error.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: lib/utils/caching.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const APP_CACHE_KEY = "app-cache-violet";
+export const WORKSPACE_ID_COOKIE_NAME = `${APP_CACHE_KEY}-workspace-id`;
```
Comment (nickytonline) [inline lib/utils/caching.ts:2]: I made this change to invalidate the current cookie, and also tie it to the app cache whenever we need to rotate that.

Comment (nickytonline) [reply lib/utils/caching.ts:2]: If you think it's overkill to link these two @brandonroberts @jpmcb, I can just keep them separate.

```suggestion
export const WORKSPACE_ID_COOKIE_NAME = "workspace-id";
```

Comment (brandonroberts) [reply lib/utils/caching.ts:2]: I'm good with how it its, I don't have a strong opinion about it

File: lib/utils/caching.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const APP_CACHE_KEY = "app-cache-violet";
```
Comment (jpmcb) [inline lib/utils/caching.ts:1]: +1 to moving this to a central spot 👏

**Review by jpmcb — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**